applications/Preprocessor/CMakeLists.txt

```diff
@@ -1,44 +1,46 @@
-include_directories(. ${hpGEM_SOURCE_DIR}/kernel/)
 
-#the preprocessor does not perform floating poing arithmetic, so it doesn't need special Petsc/not petsc guards
-add_executable(Preprocessor.out
-               main.cpp
-               centaur.cpp
-               hpgem.cpp
-               ../../kernel/Logger.cpp
-               ../../kernel/Base/CommandLineOptions.cpp
-               ../../kernel/Base/MpiContainer.cpp)
-
-add_executable(ExtractSurface.out
-               extractSurface.cpp
-               centaur.cpp
-               hpgem.cpp
-               ../../kernel/Logger.cpp
-               ../../kernel/Base/CommandLineOptions.cpp
-               ../../kernel/Base/MpiContainer.cpp)
-
-set(CURRENT_TARGET Preprocessor.out)
-include(${CMAKE_SOURCE_DIR}/conf/cmake/CLionFix.cmake)
-set(CURRENT_TARGET ExtractSurface.out)
-include(${CMAKE_SOURCE_DIR}/conf/cmake/CLionFix.cmake)
-set(CURRENT_TARGET)
+add_library(Preprocessor-lib STATIC
+lib/centaur.cpp
+lib/hpgem.cpp
+lib/gmsh.cpp
+	)
+	# TODO: Lacking source/include separation
+	target_include_directories(Preprocessor-lib
+		PUBLIC
+			${CMAKE_CURRENT_SOURCE_DIR}/lib
+	)
+	target_link_libraries(Preprocessor-lib
+		PUBLIC
+            HPGEM::HPGEM)
+            
 
 if(hpGEM_USE_MPI)
-    target_link_libraries(Preprocessor.out MPI::MPI_CXX)
-    target_link_libraries(ExtractSurface.out MPI::MPI_CXX)
+    target_link_libraries(Preprocessor-lib PUBLIC MPI::MPI_CXX)
 endif()
 
 if(hpGEM_USE_METIS)
-    target_link_libraries(Preprocessor.out METIS::METIS)
-    target_link_libraries(ExtractSurface.out METIS::METIS)
+    target_link_libraries(Preprocessor-lib PUBLIC METIS::METIS)
 endif()
 
 if(hpGEM_USE_ANY_PETSC)
-    target_link_libraries(Preprocessor.out PETSc::PETSc)
-    target_link_libraries(ExtractSurface.out PETSc::PETSc)
+    target_link_libraries(Preprocessor-lib PUBLIC PETSc::PETSc)
 endif()
 
 if(hpGEM_USE_SLEPC)
-    target_link_libraries(Preprocessor.out SLEPc::SLEPc)
-    target_link_libraries(ExtractSurface.out SLEPc::SLEPc)
+    target_link_libraries(Preprocessor-lib PUBLIC SLEPc::SLEPc)
 endif()
+
+#the preprocessor does not perform floating poing arithmetic, so it doesn't need special Petsc/not petsc guards
+add_executable(Preprocessor.out
+               main.cpp
+               ../../kernel/Logger.cpp
```

Comment:
If you switch to linking to hpgem, why then still include these source files?

Reply:
I do not link to HPGEM, only to a small part of the kernel files. It was sort of like this before, so I figured I keep it. BUt I do not care either way.